### PR TITLE
Vacillate

### PR DESCRIPTION
**Pull request by pthariensflame**
The coproduct of `Semigroup`s

### CODE REVIEW

**Comment by CodaFi** [comment]
:heart:

File: Swiftz/Semigroup.swift [hunk 1]
```diff
@@ -54,3 +54,40 @@ public struct Max<A: Comparable> : Semigroup {
 		}
 	}
 }
+
+/// The coproduct of `Semigroup`s
+public struct Vacillate<A : Semigroup, B : Semigroup> : Semigroup {
+	public let values : [Either<A, B>] // this array will never be empty
+
+	public init(_ vs : [Either<A, B>]) {
+		assert(!vs.isEmpty, "Cannot construct a \(Vacillate<A, B>.self) with no elements.")
```
Comment (CodaFi) [inline Swiftz/Semigroup.swift]: This assert will go away in release builds.  Maybe

```if vs.isEmpty {
    error("...")
}
```

Comment (pthariensflame) [inline Swiftz/Semigroup.swift]: I'll change it, then.

**Comment by CodaFi** [comment]
:boat:

**Pull request merged**